File: Android/SayHi/app/src/main/java/de/oklemenz/sayhi/model/NewCategory.java
package de.oklemenz.sayhi.model;

import java.io.Serializable;

import de.oklemenz.sayhi.service.Utilities;

/**
 * Created by Oliver Klemenz on 22.02.17.
 */

public class NewCategory implements Serializable {

    public String key = "";
    public String langCode = "";
    private String name = "";

    public void setName(String name) {
        this.name = Utilities.capitalize(name);
    }

    public String getName() {
        return name;
    }

    public Category primaryLangCategory;

    public Category getCategory() {
        return new Category(
                key,
                langCode,
                name,
                Category.CategoryStagedColor,
                Category.CategoryStagedIcon,
                0,
                primaryLangCategory != null ? primaryLangCategory.key : "",
                null,
                null);
    }
}

File: iOS/SayHi/SayHi/NewCategory.swift
//
//  NewCategory.swift
//  SayHi
//
//  Created by Klemenz, Oliver on 19.12.16.
//  Copyright © 2016 Klemenz, Oliver. All rights reserved.
//

import Foundation

class NewCategory {
    
    var key: String = ""
    var langCode: String = ""
    var _name: String = ""
    var name: String {
        set {
            _name = newValue.capitalize
        }
        get { return _name }
    }
    var primaryLangCategory: Category?
    
    var category: Category {
        return Category(key: key,
                        langCode: langCode,
                        name: name,
                        color: CategoryStagedColor,
                        icon: CategoryStagedIcon,
                        order: 0,
                        primaryLangKey: primaryLangCategory?.key ?? "",
                        refKey: nil,
                        refPrimaryLangKey: nil)
    }
}
